lib/libc/math: Add lrint, lrintf and lrintl API implementation to libc

neel-samsung opened the pull request:
Add lrint, lrintf and lrintl API implementation to libc. These APIs shall round their argument to the nearest integer value, rounding according to the current rounding direction.

sunghan-chang reviewed (commented):
1. Please fix violations in coding rules.
2. Could you add test result at commit description?
    1. How will you verify your implementation? Will make UTCs for them?

neel-samsung commented:
> 1. Please fix violations in coding rules.
> 2. Could you add test result at commit description?
>    
>    1. How will you verify your implementation? Will make UTCs for them?

Thank you for Review.

1. I have fixed coding guideline violation. Please let me know if I have missed any.
2. I have also added Kernel TCs for these APIs Testing and added the result in commit description.

sunghan-chang commented:
@neel-samsung You should not mix different topics in a single commit. Please split adding libc and adding tc for them.

edwakuwaku commented on lib/libc/math/lib_lrint.c:
```diff
@@ -0,0 +1,79 @@
+	double intpart, fracpart;
+
+	/* If x is NaN, infinite, -infinite, too large or too small, domain error is returned */
+
```
It seems added many unnecessary spaces

sunghan-chang commented on lib/libc/math/lib_lrint.c:
```diff
@@ -0,0 +1,79 @@
+
+	/* If x is NaN, infinite, -infinite, too large or too small, domain error is returned */
+
+	if (isnan(x) || isinf(x) || x > (double) LONG_MAX || x < (double) LONG_MIN) {
```
Almost type casting has a space. Let's remove it from all.

sunghan-chang commented:
@neel-samsung Could you check comment to merge by today?

edwakuwaku commented on lib/libc/math/lib_lrintf.c:
```diff
@@ -0,0 +1,77 @@
+long int lrintf(float x) 
+{
+	long int ret;
+	float intpart, fracpart;
```
similar here
float intpart;
float fracpart;

neel-samsung commented:
> @neel-samsung Could you check comment to merge by today?

Okay, I will update it by today.

edwakuwaku commented on lib/libc/math/lib_lrintf.c:
```diff
@@ -0,0 +1,70 @@
+	ret = (long int)floorf(x);
+#else
+	fracpart = modff(x, &intpart);	/* Separate the integer and fractional parts */
+	if(x >= 0.0) {
```
if (x >= 0.0)

edwakuwaku commented on lib/libc/math/lib_lrintl.c:
```diff
@@ -0,0 +1,72 @@
+	ret = (long int)floorl(x);
+#else
+	fracpart = modfl(x, &intpart);	/* Separate the integer and fractional parts */
+	if(x >= 0.0) {
```
if (x >= 0.0)

kishore-sn commented on lib/libc/math/lib_lrint.c:
```diff
@@ -0,0 +1,72 @@
+ * Public Functions
+ ************************************************************************/
+
+#ifdef CONFIG_HAVE_DOUBLE
```
Let us add #include <tinyara/config.h>as the first include file always to get the config values.
